src/ch.usi.inf.nodeprof/src/ch/usi/inf/nodeprof/handlers/BuiltinRootEventHandler.java

```diff
@@ -33,7 +34,11 @@ public BuiltinRootEventHandler(EventContext context) {
     }
 
     public Object getReceiver(VirtualFrame frame) {
-        return frame.getArguments()[0];
+        Object receiver = frame.getArguments()[0];
+        if (receiver == JSFunction.CONSTRUCT) {
```

Comment:
I am always a bit concerned when we rely on such internal impl details, but I agree that for now this is the only solution.